govcd/metadata_test.go

```diff
@@ -898,3 +899,47 @@ func (vcd *TestVCD) Test_MetadataOnVdcNetworkCRUD(check *C) {
 	check.Assert(metadata, NotNil)
 	check.Assert(len(metadata.MetadataEntry), Equals, 0)
 }
+
+func (vcd *TestVCD) Test_MetadataByHrefCRUD(check *C) {
+	fmt.Printf("Running: %s\n", check.TestName())
+	storageProfileRef, err := vcd.vdc.FindStorageProfileReference(vcd.config.VCD.StorageProfile.SP1)
+	if err != nil {
+		check.Skip(fmt.Sprintf("Test_MetadataByHrefCRUD: Storage Profile %s not found. Test can't proceed", vcd.config.VCD.StorageProfile.SP1))
+		return
+	}
+
+	storageProfileAdminHref := strings.ReplaceAll(storageProfileRef.HREF, "api", "api/admin")
```

Comment:
Is it only 'admin' which can operate on metadata? What could happen if I "set" on the url with `"api/admin"` and read on non-admin?

Reply:
In this case, for `Storage Policy`, yes, it's only the admin who can get, put, post and delete metadata. I will add also a test to get metadata with **no admin** user, good point (it's the only thing the tenant user can do for storage profile metadata)